src/sdk/VoiceProfileEnrollmentResult.ts

```diff
@@ -28,48 +28,111 @@ export interface IEnrollmentResultDetails {
+        const reason: ResultReason = json.status === "succeeded" ? ResultReason.EnrolledVoiceProfile : ResultReason.Canceled;
+        const result = new VoiceProfileEnrollmentResult(reason, null, null);
+        result.privDetails = this.getIdentificationDetails(json.processingResult, profileId);
+        return result;
+    }
+
+    public static FromIdentificationProfileList(array: any[]): VoiceProfileEnrollmentResult[] {
+        const results: VoiceProfileEnrollmentResult[] = [];
+        for (const item of array) {
+            const reason: ResultReason = item.enrollmentStatus.toLowerCase() === "enrolling" ?
+                ResultReason.EnrollingVoiceProfile : item.enrollmentStatus.toLowerCase() === "enrolled" ?
+                ResultReason.EnrolledVoiceProfile : ResultReason.Canceled;
+            const result = new VoiceProfileEnrollmentResult(reason, null, null);
+            result.privDetails = this.getIdentificationDetails(item);
+            results.push(result);
+        }
+        return results;
+    }
+
+    public static FromVerificationProfileList(array: any[]): VoiceProfileEnrollmentResult[] {
+        const results: VoiceProfileEnrollmentResult[] = [];
+        for (const item of array) {
+            const reason: ResultReason = item.enrollmentStatus.toLowerCase() === "enrolling" ?
+                ResultReason.EnrollingVoiceProfile : item.enrollmentStatus.toLowerCase() === "enrolled" ?
+                ResultReason.EnrolledVoiceProfile : ResultReason.Canceled;
+            const result = new VoiceProfileEnrollmentResult(reason, null, null);
+            result.privDetails = this.getVerificationDetails(item);
+            results.push(result);
+        }
+        return results;
+
+    }
+
     public get reason(): ResultReason {
         return this.privReason;
     }
 
-    public get enrollmentsCount(): number {
-        return this.privDetails.enrollmentsCount;
+    public get enrollmentCount(): number {
+        return this.privDetails.enrollmentCount;
     }
 
-    public get enrollmentsLength(): number {
-        return this.privDetails.enrollmentsLength;
+    public get enrollmentLength(): number {
+        return this.privDetails.enrollmentLength;
     }
 
     public get properties(): PropertyCollection {
         return this.privProperties;
     }
 
-    public get enrollmentResultDetails(): IEnrollmentResultDetails {
+    public get enrollmentResultDetails(): EnrollmentResultDetails {
         return this.privDetails;
     }
 
     public get errorDetails(): string {
         return this.privErrorDetails;
     }
+
+    private static getIdentificationDetails(json: any, profileId?: string): any {
+        return {
+            audioSpeechLength: json.speechTime ? parseFloat(json.speechTime) : null,
+            enrollmentLength: parseFloat(json.enrollmentSpeechTime),
```

Comment:
enrollmentLength and remainingEnrollmentSpeechLength - we should guard against NaN on those as well? Or do we necessarily need to? In any case, I'm wondering if all or none should, for consistency